src/zope/sqlalchemy/datamanager.py

```diff
@@ -84,22 +85,18 @@ def __init__(
             self, session, status, transaction_manager, keep_session=False):
         self.transaction_manager = transaction_manager
 
-        # Support SQLAlchemy 2.0
-        # https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=get_transaction#sqlalchemy.orm.Session.get_transaction
-        transaction = (
-            session.get_transaction()
-            if hasattr(session.transaction, "get_transaction")
-            else session.transaction
-        )
-
-        # Support both SQLAlchemy 1.0 and 1.1
-        # https://github.com/zopefoundation/zope.sqlalchemy/issues/15
-        _iterate_parents = (
-            getattr(transaction, "_iterate_self_and_parents", None)
-            or transaction._iterate_parents
-        )
-
-        self.tx = _iterate_parents()[-1]
+        if sqlalchemy_version >= "1.4.0":
```

Comment:
This is not the right way to compare versions.  Consider, for example, sqlalchemy_version of `"1.10.0"`.